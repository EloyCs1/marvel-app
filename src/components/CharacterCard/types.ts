import { AppCharacter } from "../../service/types";

export interface CharacterCardProps {
  character: AppCharacter.Character;
}
