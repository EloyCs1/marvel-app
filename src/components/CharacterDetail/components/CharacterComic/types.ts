import { AppCharacter } from "src/service/types";

export interface CharacterComicProps {
  comic: AppCharacter.Comic;
}
