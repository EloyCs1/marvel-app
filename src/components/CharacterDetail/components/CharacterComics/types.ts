import { AppCharacter } from "src/service/types";

export interface CharacterComicsProps {
  comics: AppCharacter.Comic[];
  loading: boolean;
}
