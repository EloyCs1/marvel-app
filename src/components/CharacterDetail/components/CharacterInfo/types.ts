import { AppCharacter } from "src/service/types";

export interface CharacterInfoProps {
  detail: AppCharacter.Character;
  loading: boolean;
}
