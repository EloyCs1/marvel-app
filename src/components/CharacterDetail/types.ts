export interface CharacterDetailProps {
  characterId: number;
}
