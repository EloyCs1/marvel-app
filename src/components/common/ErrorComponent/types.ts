export interface ErrorComponentProps {
  testId?: string;
  tryAgain?: () => void;
}
