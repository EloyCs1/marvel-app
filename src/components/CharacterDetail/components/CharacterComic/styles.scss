.characterComic {
  display: grid;
  gap: 8px;
  padding-bottom: 24px;
  width: 179px;

  &__image {
    background-size: cover;
    background-repeat: no-repeat;
    height: 275px;
    width: inherit;
  }

  @media (max-width: 960px) {
    width: 169px;
    &__image {
      height: 260px;
    }
  }

  @media (max-width: 575px) {
    width: 149px;
    &__image {
      height: 229px;
    }
  }

  &__title {
    font-weight: 500px;
    line-height: 19px;
  }

  &__onsaleDate {
    font-size: 12px;
    line-height: normal;
  }
}
