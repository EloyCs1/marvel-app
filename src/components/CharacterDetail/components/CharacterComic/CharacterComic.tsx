import { EMPTY_VALUE } from "src/service/constants";

import { CharacterComicProps } from "./types";
import "./styles.scss";

const CharacterComic: React.FC<CharacterComicProps> = ({ comic }) => {
  const { id, image, title, onsaleDate } = comic;
  const testId = `characterComic-${id}`;
  return (
    <div className="characterComic" data-testid={testId}>
      <div
        className="characterComic__image"
        style={{ backgroundImage: `url(${image})` }}
        data-testid={`${testId}-image`}
      />
      <p className="characterComic__title" data-testid={`${testId}-title`}>
        {title}
      </p>
      <p className="characterComic__onsaleDate" data-testid={`${testId}-onsaleDate`}>
        {isNaN(onsaleDate) ? EMPTY_VALUE : onsaleDate}
      </p>
    </div>
  );
};
export default CharacterComic;
