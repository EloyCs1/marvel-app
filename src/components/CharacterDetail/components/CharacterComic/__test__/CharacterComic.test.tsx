import { render } from "@testing-library/react";
import { COMICS } from "src/mocks/mocks";
import { AppWrapper } from "src/test/test.utils";
import { CharacterComicProps } from "../types";
import CharacterComic from "../CharacterComic";

const defaultProps: CharacterComicProps = {
  comic: COMICS[0],
};

describe("CharacterComic", () => {
  test("Default render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterComic {...defaultProps} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("Empty onsaleDate render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterComic comic={{ ...defaultProps.comic, onsaleDate: NaN }} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
