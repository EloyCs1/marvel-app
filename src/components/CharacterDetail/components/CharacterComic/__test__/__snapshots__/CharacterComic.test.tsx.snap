// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterComic Default render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterComic"
        data-testid="characterComic-40632"
      >
        <div
          class="characterComic__image"
          data-testid="characterComic-40632-image"
          style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/2/00/5ba3bfcc55f5a.jpg);"
        />
        <p
          class="characterComic__title"
          data-testid="characterComic-40632-title"
        >
          Hulk (2008) #53
        </p>
        <p
          class="characterComic__onsaleDate"
          data-testid="characterComic-40632-onsaleDate"
        >
          2012
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterComic"
      data-testid="characterComic-40632"
    >
      <div
        class="characterComic__image"
        data-testid="characterComic-40632-image"
        style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/2/00/5ba3bfcc55f5a.jpg);"
      />
      <p
        class="characterComic__title"
        data-testid="characterComic-40632-title"
      >
        Hulk (2008) #53
      </p>
      <p
        class="characterComic__onsaleDate"
        data-testid="characterComic-40632-onsaleDate"
      >
        2012
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterComic Empty onsaleDate render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterComic"
        data-testid="characterComic-40632"
      >
        <div
          class="characterComic__image"
          data-testid="characterComic-40632-image"
          style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/2/00/5ba3bfcc55f5a.jpg);"
        />
        <p
          class="characterComic__title"
          data-testid="characterComic-40632-title"
        >
          Hulk (2008) #53
        </p>
        <p
          class="characterComic__onsaleDate"
          data-testid="characterComic-40632-onsaleDate"
        >
          -
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterComic"
      data-testid="characterComic-40632"
    >
      <div
        class="characterComic__image"
        data-testid="characterComic-40632-image"
        style="background-image: url(http://i.annihil.us/u/prod/marvel/i/mg/2/00/5ba3bfcc55f5a.jpg);"
      />
      <p
        class="characterComic__title"
        data-testid="characterComic-40632-title"
      >
        Hulk (2008) #53
      </p>
      <p
        class="characterComic__onsaleDate"
        data-testid="characterComic-40632-onsaleDate"
      >
        -
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
